ion-schema/src/main/kotlin/com/amazon/ionschema/model/SchemaDocument.kt

```diff
@@ -0,0 +1,40 @@
+package com.amazon.ionschema.model
+
+import com.amazon.ion.IonValue
+import com.amazon.ionschema.IonSchemaVersion
+
+/**
+ * Represents an Ion Schema document.
+ */
+@ExperimentalIonSchemaModel
+data class SchemaDocument internal constructor(
+    val id: String?,
+    val items: List<Item>
+) {
+    val ionSchemaVersion: IonSchemaVersion = items.firstOrNull { it !is Item.OpenContent }
+        ?.let { it as? Item.VersionMarker }
+        ?.value
+        ?: IonSchemaVersion.v1_0
```

Comment:
I don't think there is any invariant that requires validation here.

From the [ISL Versioning Spec](https://amazon-ion.github.io/ion-schema/docs/isl-versioning#ion-schema-version-markers)
> The first value in the schema SHOULD be an ISL Version Marker; if a top-level ISL value is encountered before encountering an ISL version marker, that schema SHALL use ISL 1.0.

So, we're finding the first this that is not open content (i.e. is is a top-level ISL value). If that top-level value is a version marker, then we use that version. Otherwise, it is ISL 1.0.